### PR TITLE
[3851][16.0][ADD] purchase_order_no_owner

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[3851](https://www.quartile.co/web#id=3851&cids=3&model=project.task&view_type=form)

### CODE REVIEW

File: purchase_order_no_owner/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    no_owner = fields.Boolean()
+
+    @api.constrains("no_owner", "owner_id")
+    def _check_owner_id(self):
+        for record in self:
+            if not record.no_owner and not record.owner_id:
+                raise UserError(
+                    _(
+                        "Please select the owner. if you don't want to select any owner for"
+                        " this order, you can set no owner field as True."
```
Comment (yostashiro) [inline purchase_order_no_owner/models/purchase_order.py]:
```suggestion
                        " this order, you can set No Owner field as True."
```

File: purchase_order_no_owner/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+This module adds no_owner field to purchase order.
```
Comment (yostashiro) [inline purchase_order_no_owner/readme/DESCRIPTION.rst]: Please add the business context.

**Review by rinaldifirdaus — APPROVED** [review]

File: purchase_order_no_owner/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+from odoo.exceptions import UserError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    no_owner = fields.Boolean()
+
+    @api.constrains("no_owner", "owner_id")
+    def _check_owner_id(self):
```
Comment (yostashiro) [inline purchase_order_no_owner/models/purchase_order.py]: @rinaldifirdaus I think we should actually do this check when confirming PO instead.  Making it a constraint may introduce an inconvenience of having to be sure about the arrangement at the very beginning.  Users may just select No Owner when they are not sure when they create a draft PO, and then later forget to update the settings even when it later turns out an owner should be assigned.

File: purchase_order_no_owner/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,12 @@
+This module adds no_owner field in the purchase order,
+allowing users to indicate that the ordered inventory is not associated with any specific owner.
```
Comment (yostashiro) [inline purchase_order_no_owner/readme/DESCRIPTION.rst]: More emphasis should be put on preventing the oversight than giving options to users.

File: purchase_order_no_owner/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+Background
+~~~~~~~~~~
+
+users may unintentionally overlook filling out the owner_id field when creating a purchase order
+for inventory items that should be linked to a specific owner and subsequently confirm the order.
+Once the PO confirmed, users will not able to edit the owner_id field to fill the correct owner.
+
+This module is designed to address this situation by Implementing a process that blocks saving the order or confirming it if the owner_id is blank.
+Additionally, when the order is not associated with any specific owner, users have the option to set the no_owner field to "True" instead.
```
Comment (yostashiro) [inline purchase_order_no_owner/readme/DESCRIPTION.rst]: This is not part of the background, therefore move it to the main part of the description or USAGE.rst.

File: purchase_order_no_owner/readme/DESCRIPTION.rst [hunk 6]
```diff
@@ -0,0 +1,10 @@
+This module is designed to prevent users from accidentally forgetting to input owner information on a purchase order
+by implementing an order confirmation prevention if the owner_id is not filled in.
+If the order is not associated with any specific owner, users have the option to set the no_owner field to "True" instead.
+
+Background
+~~~~~~~~~~
+
+users may unintentionally overlook filling out the owner_id field when creating a purchase order
```
Comment (yostashiro) [inline purchase_order_no_owner/readme/DESCRIPTION.rst]:
```suggestion
Users may unintentionally overlook filling out the owner_id field when creating a purchase order
```

File: purchase_order_no_owner/readme/DESCRIPTION.rst [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+for inventory items that should be linked to a specific owner and subsequently confirm the order.
+Once the PO is confirmed, users will not be able to edit the owner_id field to fill in the correct owner.
```
Comment (yostashiro) [inline purchase_order_no_owner/readme/DESCRIPTION.rst]:
```suggestion
Once the PO is confirmed, users are not be able to edit the owner_id field to fill in the correct owner.
```

**Review by yostashiro — APPROVED** [review]
Thank you.

File: purchase_order_no_owner/readme/DESCRIPTION.rst [hunk 8]
```diff
@@ -0,0 +1,10 @@
+This module is designed to prevent users from accidentally forgetting to input owner information on a purchase order
+by implementing an order confirmation prevention if the owner_id is not filled in.
+If the order is not associated with any specific owner, users have the option to set the no_owner field to "True" instead.
+
+Background
+~~~~~~~~~~
+
+Users may unintentionally overlook filling out the owner_id field when creating a purchase order
+for inventory items that should be linked to a specific owner and subsequently confirm the order.
+Once the PO is confirmed, users are not be able to edit the owner_id field to fill in the correct owner.
```
Comment (yostashiro) [inline purchase_order_no_owner/readme/DESCRIPTION.rst:10]:
```suggestion
Once the PO is confirmed, users are not able to edit the owner_id field to fill in the correct owner.
```

@rinaldifirdaus I tend to make quick/partial suggestions, so please make sure to check the comleteness/comprehensiveness yourself upon update.

**Review by kanda999 — APPROVED** [review]
LGTM 👍

**Pull request merged**